### PR TITLE
Adding basic gRPC layer

### LINKED ISSUE #19: Complete basic service API impl
Both on the `Service` type and on the gRPC server.

### PR DESCRIPTION
**Pull request by moshababo**
Closes #19.

### CODE REVIEW

File: rpc/rpcserver.go [hunk 1]
```diff
@@ -18,10 +20,71 @@ var _ api.PoetServer = (*rpcServer)(nil)
...
+	out := new(api.GetMembershipProofResponse)
+	out.MerkleProof = proof
+	return out, nil
+}
+
+func nativeLabelsToWire(native [shared.T]shared.Labels) (wire []*api.Labels) {
+	for _, labels := range native {
+		var labelsMsg api.Labels
+		for _, label := range labels {
+			labelsMsg.Labels = append(labelsMsg.Labels, label)
```
Comment (antonlerner) [inline rpc/rpcserver.go:85]: you can append a slice to a slice in go

File: rpccore/rpcserver.go [hunk 2]
```diff
@@ -193,30 +192,27 @@ func wireLabelsToNative(in []*api.Labels) (native [shared.T]shared.Labels) {
...
 }
 
-// TODO: find a better way to do this
-func wireChallengeToNative(in []string) (native [shared.T]shared.Identifier) {
-	for i, identifier := range in {
+func wireChallengeToNative(wire []string) (native [shared.T]shared.Identifier) {
+	for i, identifier := range wire {
 		native[i] = shared.Identifier(identifier)
 	}
 	return native
```
Comment (antonlerner) [inline rpccore/rpcserver.go]: since your output parameter is named you can just use `return` to return `native`

File: server.go [hunk 3]
```diff
@@ -41,12 +41,12 @@ func startServer() error {
 		proxyRegstr = append(proxyRegstr, apicore.RegisterPoetCoreProverHandlerFromEndpoint)
 		proxyRegstr = append(proxyRegstr, apicore.RegisterPoetVerifierHandlerFromEndpoint)
 	} else {
-		s, err := service.NewService(cfg.Service)
+		service, err := service.NewService(cfg.Service)
 		if err != nil {
 			return nil
```
Comment (antonlerner) [inline server.go]: return error

File: rpc/rpcserver.go (hunk 1 above)
Comment (moshababo) [reply rpc/rpcserver.go:85]: The slice item type is different, so I didn't find a way to do it without iterating through the items. LMK if you have a better way to implement this.

**Comment by antonlerner** [comment]
> Closes #19.

This PR does not address concurrency issues... is it not part of https://github.com/spacemeshos/poet-ref/issues/19?

**Comment by moshababo** [comment]
@antonlerner Yes, i'll handle it with other things in the next iteration, after finishing the NIPST construction in the full-node.

**Pull request merged**